### DISCUSSION TITLE
Jupyterlite extension build failure

### DISCUSSION THREAD

**Discussion by ashok777** [jupyterlite/jupyterlite#1304] [Q&A]
I am creating the necessary environment to build/deploy a jupyterlite server extension and get to the step where I need to run setup.py to build the extension, i.e python -m pip install -e . The step fails with error:

Building wheels for collected packages: pyzmq
Building wheel for pyzmq (pyproject.toml): started
Building wheel for pyzmq (pyproject.toml): finished with status 'error'

The extension was originally built with python 3.9 and has been working successfully. The failure seen now is 
with python 3.12.1.  Looking for suggestion on what my recouse is to get a successful build

**Comment by jtpio** [comment]
Thanks @ashok777.

It does seem to be an issue with `pyzmq` and Python 3.12. Maybe a wheel not available for your system? 

There were a few other issues reported on other repos that may be related, for example: https://github.com/jupyterlab/jupyterlab/issues/15332. Although not sure what the status on this is at the moment.

**Reply by ashok777** [reply]
@jtpio Thanks for your response. I am perfectly content with using jupyterlite 1.0 and python 3.10 as I am only using the javascript kernel notebooks and the environment I have works well for that usage. So for now I am abandoning plans of upgrading to the latest jupyterlite and python 3.12. I might try again and upgrade to a more updated version at a later time.
